tests/integration/models/api-key.spec.ts

```diff
@@ -124,6 +124,31 @@ describe('API Key model', function () {
 		});
 	});
 
+	describe('balena.models.apiKey.getProvisioningApiKeysByApplication', function () {
+		givenLoggedInUser(before);
+
+		before(function () {
+			return balena.models.application
+				.generateApiKey('SampleOrg/sampleApp')
```

Comment:
It doesn't look like this application is created anywhere, so I expect this to reject in all cases.
You should add a `givenAnApplication(before);` right after the `givenLoggedInUser` and then in here use 
```suggestion
				.generateApiKey(this.application.id)
```